sources/web/datalab/idleTimeout.ts

```diff
@@ -0,0 +1,256 @@
+/*
+ * Copyright 2017 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+/// <reference path="common.d.ts" />
+
+import childProcess = require('child_process');
+import http = require('http');
+import logging = require('./logging');
+import net = require('net');
+import querystring = require('querystring');
+import url = require('url');
+import userManager = require('./userManager');
+
+const idleCheckIntervalSeconds = 5;    // Seconds to wait between idle checks
+
+let idleTimeoutEnabled = true;  // Allow for explicit disable without clearing other values
+let idleTimeoutSeconds = 0;   // Shutdown after being idle this long; disabled if 0 or NaN
+let lastUserActivity: number;  // The epoch, in seconds, of the last user activity
```

Comment:
I renamed this variable to be more reflective of reality. This module doesn't define what user activity is.